ido/tests/sim_test/ido_test/whitelist_test.rs

```diff
@@ -0,0 +1,217 @@
+use crate::utils::init;
+use near_sdk::serde_json::json;
+use near_sdk::json_types::{U64, U128};
+use near_sdk_sim::{DEFAULT_GAS, to_yocto};
+use kulapad_ido::modules::project::{ProjectJson,ProjectStatus,WhitelistType};
+use kulapad_ido::modules::account::ProjectAccountJson;
+use kulapad_ido::staking_contract::AccountJson;
+
+#[test]
+pub fn test_join_whitelist(){
+    let (root,alice,ido,_,_,ido_contract) = init();
+    
+    ido.call(
+        ido_contract.account_id(),
+        "update_project_whitelist_date",
+        &json!({
+            "project_id" : 1,
+        }).to_string().as_bytes(),
+        DEFAULT_GAS,
+        0
+    );
+
+    ido.call(
+        ido_contract.account_id(),
+        "update_project_status",
+        &json!({
+            "project_id" : 1,
+            "new_status": ProjectStatus::Whitelist
+        }).to_string().as_bytes(),
+        DEFAULT_GAS,
+        0
+    );
+
+    let project_json : ProjectJson = root.view(
+        ido_contract.account_id(),
+        "get_project",
+        &json!({
+            "project_id" : 1
+        }).to_string().as_bytes()
+    ).unwrap_json();
+    println!("{:?}",project_json);
+    assert_eq!(project_json.whitelist_type,WhitelistType::None,"Project 1 WhitelistType not match");
+
+    let current_time : U64 = root.view(
+        ido_contract.account_id(),
+        "get_current_block_timestamp",
+        &json!({}).to_string().as_bytes()
+    ).unwrap_json();
+    println!("Current time: {}",current_time.0);
+
+    let mut is_whitelisted : bool = root.view(
+        ido_contract.account_id(),
+        "is_whitelist",
+        &json!({
+            "project_id" : 1,
+            "account_id": alice.account_id()
+        }).to_string().as_bytes()
+    ).unwrap_json();
+
+    
+    assert_eq!(is_whitelisted,false, "Joined whitelist when not call");
+
+    alice.call(
+        ido_contract.account_id(),
+        "internal_register_whitelist",
```

Comment:
The simulation tests should call the public function, not an internal one. Let's think that simulation testing simulates the behaviors of the user who cannot access the internal contract functions. Please use the public version of this function.